sub/filter_sdh.c

```diff
@@ -84,26 +77,33 @@ static int get_char_bytes(char *str)
     return 0;
 }
 
-static const char *get_right_enclosure(char *left)
+static bool is_parenthesis_pair(const char *left, const char *right)
 {
-    // See if the right hand character is mapped. If not, just return the same thing.
-    for (int i = 0; enclosure_pair[i][0]; i++) {
-        if (strcmp(left, enclosure_pair[i][0]) == 0)
-            return enclosure_pair[i][1];
-    }
-    return left;
+    if (!strcmp(left, "(") && !strcmp(right, ")"))
+        return true;
+    if (!strcmp(left, "\uFF08") && !strcmp(right, "\uFF09"))
+        return true;
+    return false;
 }
 
-static bool valid_left_enclosure(struct sd_filter *sd, char *str)
+static int matching_enclosure(char **enclosures, char *str)
 {
-    // All characters in this string are valid left hand enclosure characters.
-    char *enclosures = sd->opts->sub_filter_SDH_enclosures;
-    int len = strlen(enclosures);
-    for (int i = 0; i < len; i++) {
-        if (str && str[0] && str[0] == enclosures[i])
-            return true;
+    int str_bytes = get_char_bytes(str);
+    if (!str_bytes)
+        goto done;
+    char enclosure[5] = {0};
+    for (int i = 0; i < str_bytes; ++i)
+        enclosure[i] = str[i];
+    for (int i = 0; enclosures && enclosures[i]; i++) {
+        int match_bytes = get_char_bytes(enclosures[i]);
+        char match[5] = {0};
+        for (int j = 0; j < match_bytes; ++j)
+            match[j] = enclosures[i][j];
+        if (str && strcmp(enclosure, match) == 0)
+            return i;
     }
-    return false;
+done:
+    return -1;
 }
```

Comment:
Is there a good reason why this function cannot be simplified like this? (untested)
```c
// return the index in enclosures where the opening codepoint
// is the same as the first codepoint in str, or -1 if not found
static int matching_enclosure(char **enclosures, char *str)
{
    int len = get_char_bytes(str);
    for (int i = 0; len && enclosures[i]; ++i)
        if (!strncmp(str, enclosures[i], len))
            return i;
    return -1;
}
```